packages/core/src/dispatcher.ts

```diff
@@ -284,18 +278,15 @@ export default class Dispatcher extends EventEmitter {
    * @private
    */
   async _handleUpdateMessage(message: any): Promise<void> {
-    // TODO Add validation the message adheres to the proper format.
-
     const { doc, tip } = message
-    if (!this._documents[doc]) {
-      return
+    const docId = DocID.fromString(doc)
+    if (await this.repository.has(docId)) {
```

Comment:
Can we invert this if-statement and do an early-return if we don't care about the message, as was done previously?  I'm a big fan of early-returns rather than introducing additional layers of indentation and scope nesting.  I find the code tends to be easier to read and less error-prone.

Here and in the other handle*Message functions below